src/common/framework/tree.ts

```diff
@@ -377,3 +398,44 @@ function insertLeaf(parent: TestSubtree, query: TestQuerySingleCase, t: RunCase)
   assert(!parent.children.has(key));
   parent.children.set(key, leaf);
 }
+
+function dissolveLevelBoundaries(tree: TestTreeNode): TestTreeNode {
+  if ('children' in tree) {
+    if (tree.children.size === 1 && tree.description === undefined) {
+      // Loops exactly once
+      for (const [, child] of tree.children) {
+        if (child.query.level > tree.query.level) {
+          const newtree = dissolveLevelBoundaries(child);
+
+          // If newtree is a leaf (case), make sure it has a readable name.
+          if ('run' in newtree) {
+            newtree.readableRelativeName = readableNameForCase(newtree.query);
+          }
+
+          return newtree;
+        }
+      }
+    }
+
+    for (const [k, child] of tree.children) {
+      const newChild = dissolveLevelBoundaries(child);
+      if (newChild !== child) {
+        tree.children.set(k, newChild);
+      }
+    }
+  }
+  return tree;
+}
+
+/** Generate a readable relative name for a case (used in standalone). */
+function readableNameForCase(query: TestQuerySingleCase): string {
+  let name: string;
+  const paramsKeys = Object.keys(query.params);
+  if (paramsKeys.length === 0) {
+    name = query.testPathParts[query.testPathParts.length - 1] + kBigSeparator;
+  } else {
+    const lastKey = paramsKeys[paramsKeys.length - 1];
+    name = stringifySingleParam(lastKey, query.params[lastKey]);
```

Comment:
Why do we only show a single param? Is it not possible to a single test with a single leaf node with multiple params?

Reply:
It's really subtle, but it works out because we only dissolve one level boundary (the one between multi-case and single-case), and not every single-child subtree (like I was doing before).
![image](https://user-images.githubusercontent.com/606355/82489244-2b45fb80-9a96-11ea-94c1-fdadf5e3bcfb.png)